pkg/cmdline/cmd.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cmdline
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// CommandManager holds root command or first parent
+// and can stores group of command
+type CommandManager struct {
+	rootCmd   *cobra.Command
+	groupCmds map[string][]*cobra.Command
+	errPool   []error
+	fm        *flagManager
+}
+
+// NewCommandManager instantiates a CommandManager
+func NewCommandManager(rootCmd *cobra.Command) *CommandManager {
+	if rootCmd == nil {
+		panic("nil root command passed")
+	}
+	cm := &CommandManager{
+		rootCmd:   rootCmd,
+		groupCmds: make(map[string][]*cobra.Command),
+		errPool:   make([]error, 0),
+		fm:        newFlagManager(),
+	}
+	return cm
+}
+
+func (m *CommandManager) pushError(f string, a ...interface{}) {
+	m.errPool = append(m.errPool, fmt.Errorf(f, a...))
```

Comment:
it would be better if `pushError` took error (`pushError(errs ...error)`) so that the client code can decide what to do with the errors.

If you feel formatting errors like this is necessary, you could have `pushErrorf(format string, args ...interface{})` (in addition to pushError, not instead of)

Reply:
Looks like not necessary. Will change